### PR TITLE
Add lookback for query, fix queries with small step size

### PR DESCRIPTION
**Pull request by nikunjgit**

### CODE REVIEW

File: src/query/ts/values_test.go [hunk 1]
```diff
@@ -1,134 +0,0 @@
-// Copyright (c) 2018 Uber Technologies, Inc.
```
Comment (arnikola) [inline src/query/ts/values_test.go]: Should not be deleting tests; better to replace them with tests for the new func

File: src/query/models/params.go [hunk 2]
```diff
@@ -24,6 +24,10 @@ import (
 	"time"
 )
 
+// LookbackDelta determines the time since the last sample after which a time
+// series is considered stale.
+var LookbackDelta = time.Minute
```
[the remark this answers is not in the document]
Comment (nikunjgit) [reply src/query/models/params.go]: TODO, hard to see why people would want to change it. Prom has it hardcoded.

[the remark this answers is not in the document]
Comment (benraskin92) [reply src/query/models/params.go]: understood

File: src/query/ts/values.go [hunk 3]
```diff
@@ -93,14 +90,25 @@ func (d Datapoints) AlignToBounds(bounds models.Bounds) []Datapoints {
 	stepSize := bounds.StepSize
 	t := bounds.Start
 	for i := 0; i < steps; i++ {
-		startIdx := dpIdx
-		for dpIdx < numDatapoints && d[dpIdx].Timestamp.Before(t) {
+		singleStepValues := make(Datapoints, 0)
```
Comment (benraskin92) [inline src/query/ts/values.go]: I don't like the `0`

**Pull request merged**